pkg/cmd/issue/transfer/transfer.go

```diff
@@ -66,7 +67,7 @@ func transferRun(opts *TransferOptions) error {
 		return err
 	}
 
-	destRepo, err := ghrepo.FromFullName(opts.DestRepoSelector)
+	destRepo, err := repoShared.NewRepo(opts.DestRepoSelector, opts.Config, apiClient)
```

Comment:
Note: There is an issue that already existed here where the `DestRepoSelector` can specify a host which is different from the issue host. This was outside the scope of this PR though and will be fixed in a follow up one.